omniscidb/QueryEngine/WorkUnitBuilder.cpp

```diff
@@ -327,8 +327,14 @@ void WorkUnitBuilder::process(const ir::Node* node) {
       if (scan) {
         col_var = ir::makeExpr<ir::ColumnVar>(scan->getColumnInfo(i), rte_idx);
       } else {
-        col_var = ir::makeExpr<ir::ColumnVar>(
-            node->getOutputMetainfo()[i].type(), -node->getId(), i, rte_idx, false);
+        CHECK(temporary_tables_.count(-node->getId()));
+        auto& token = temporary_tables_.at(-node->getId());
+        col_var = ir::makeExpr<ir::ColumnVar>(node->getOutputMetainfo()[i].type(),
+                                              token->dbId(),
+                                              token->tableId(),
+                                              i + 1,
```

Comment:
What is the `+1` for?

Reply:
In schemas, we always avoided zero ids for dbs/tables/cols/frags, so column_id for ResultSet column is its index + 1.

Reply:
Sounds error-prone :)

Reply:
Hmm... I can add a method to ResultSetToken to get column id by column index. Such incapsulation would make code more clear. I'm not sure this would cover all `+1` cases though.